gammapy/modeling/models/tests/test_spectral.py

```diff
@@ -440,6 +487,32 @@ def test_to_from_dict_partial_input():
     assert_allclose(actual, desired)
 
 
+def test_to_from_dict_partial_input_wrong_parameters(caplog):
+    spectrum = TEST_MODELS[0]
+    model = spectrum["model"]
+
+    model_dict = model.to_dict()
+    # Here we remove the reference energy
+    model_dict['parameters'].remove(model_dict['parameters'][0])
+    print(model_dict['parameters'][0])
+
+    model_class = SPECTRAL_MODEL_REGISTRY.get_cls(model_dict["type"])
+    new_model = model_class.from_dict(model_dict)
+
+    assert isinstance(new_model, PowerLawSpectralModel)
+
+    # actual = [par.value for par in new_model.parameters]
+    # desired = [par.value for par in model.parameters]
+    # assert_quantity_allclose(actual, desired)
+    #
+    # actual = [par.frozen for par in new_model.parameters]
+    # desired = [par.frozen for par in model.parameters]
+    # assert_allclose(actual, desired)
+
+    assert "gammapy.modeling.models.core:core.py:151 Parameter index not defined. Using default value: 2.0" in caplog.text
```

Comment:
I guess this assert should be simplified to only include the the message like "Parameter index not defined. Using default value: 2.0". Keeping the logger location will result in weird test fails, when the warning is not emitted from line 151 anymore, because code has beed added before...

Reply:
done